### DISCUSSION TITLE
daily or weekly view

### DISCUSSION THREAD

**Discussion by blondie63** [omnia-digital/livewire-calendar#21] [Q&A] [answered]
Can this package only show the monthly view or is it possible to choose a daily or weekly view?

**Comment by joshtorres** [comment, marked as answer]
It's possible to build that yourself, but we don't have an easy way to display that currently. If you do want to contribute that, we would gladly pull that into the repo

**Reply by blondie63** [reply]
@joshtorres Hi, i'd like to build and share this feature
Can you just give me a hint to understand which files to work on? I would like to copy the same functionality as fullcalendar.io ?

**Reply by joshtorres** [reply]
You would need to create some resource views for day, week, etc views. Then add the buttons to the calendar to be able to swap out the different views. 

I would probably create and pass in a dayGrid() and weekGrid() method here in LivewireCalendar. Almost everything lives in this one class, so feel free to add to it.
![image](https://github.com/omnia-digital/livewire-calendar/assets/5092957/8bbe0c95-5bf1-49a6-a868-c254be986dcd)

It depends how you want to break it down, but it might be good to have hour time slot views similar to how we have day views for the monthly view. 

I hope that helps.

**Reply by blondie63** [reply]
Hi @joshtorres thank for you answer!
I found another fork (https://github.com/rabol/livewire-calendar) that seems to have views for week and day but there are no clear examples
I've tried implementing part of that fork and I seem to be close, but it still doesn't work properly. What do you think about it ?
I attach two videos, one made with Omnia and my additions, the other native to BackpackforLaravel
what you think ?

https://github.com/omnia-digital/livewire-calendar/assets/1261851/42062d86-f1de-4140-9e73-8d697812ab94


https://github.com/omnia-digital/livewire-calendar/assets/1261851/0897b367-40f9-47a7-87dd-408b98ad7094

**Reply by joshtorres** [reply]
Yeah those week and day views look pretty good!

**Reply by blondie63** [reply]
Hi @joshtorres, Today I picked up the project again and simply did a composer update but my Calendar page now gives me this error: 
Livewire Exceptions
ComponentNotFoundException
Unable to find component: [appointments-calendar]

Has it ever happened to you? It's very strange because I haven't changed anything to my code :(

ps. i'm using "livewire/livewire": "^3.0" updated to 3.4.9

**Reply by blondie63** [reply]
I fixed it ! the problem was that last Livewire move name space to App\Livewire

then i've run "php artisan livewire:upgrade -vvv"

Now it's ok for me

**Reply by joshtorres** [reply]
oh ok thanks for letting us know.

**Reply by blondie63** [reply]
Hi @joshtorres have time for a little consultant about new feature for your livewire-calendar ?
Logically for pay :)
